### PR TITLE
feat: Decouple `extensionLayerVersion` from `addLayers`

### PR DESCRIPTION
**Pull request by samchungy**
### What does this PR do?

Resolves https://github.com/DataDog/datadog-cdk-constructs/issues/201

- Allows the extension layer to be added regardless of the `addLayers` property.

### Motivation

I would like to instrument my lambda with the node libraries without layers but the extension library is required to be added as a layer.

At the moment I use an ugly workaround to allow for this behaviour.

https://github.com/seek-oss/skuba/blob/ca287dfa624f1739fbb349bc22dfde3be9b09919/template/lambda-sqs-worker-cdk/infra/appStack.ts#L130-L139

### Testing Guidelines

- Unit Tests

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [x] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

Arguably this is a minor breaking change if the user was relying on `addLayers` to prevent layers from being added whilst specifying the extensionLayerVersion but I would hope that this is uncommon.

### Check all that apply

- [x] This PR's description is comprehensive
- [x] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [x] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,40 +209,40 @@ To further configure your Datadog construct, use the following custom parameters
...
+| `dotnetLayerVersion`         | `dotnet_layer_version`          | Version of the .NET layer to install, such as `13`. Required if you are deploying at least one Lambda function written in .NET and `addLayers` is `true`. Find the latest version number from [here][18].                                                                                                                                                                                                             |
+| `extensionLayerVersion`      | `extension_layer_version`       | Version of the Datadog Lambda Extension layer to install, such as 5. When `extensionLayerVersion` is set, `apiKey` (or if encrypted, `apiKMSKey` or `apiKeySecretArn`) needs to be set as well. When enabled, lambda function log groups will not be subscribed by the forwarder. Learn more about the Lambda extension [here][12]. If this is set, this will add a layer regardless of the `addLayer` configuration. |
```
Comment (neko-dd) [inline README.md:219]:
```suggestion
| `extensionLayerVersion`      | `extension_layer_version`       | Version of the Datadog Lambda Extension layer to install, such as 5. When `extensionLayerVersion` is set, `apiKey` (or if encrypted, `apiKMSKey` or `apiKeySecretArn`) needs to be set as well. When enabled, lambda function log groups will not be subscribed by the forwarder. Learn more about the Lambda extension [here][12]. **Note**: If this parameter is set, it adds a layer even if `addLayers` is set to `false`. |
```

Comment (neko-dd) [reply README.md:219]: (assuming that this is equivalent to the original, that is)

**Review by neko-dd — APPROVED** [review]

**Comment by samchungy** [comment]
Heyo @neko-dd and/or @astuyve or @duncanista could I trouble you for a merge/release or more feedback if this approach is particularly troubling? Cheers 😄

**Comment by neko-dd** [comment]
Docs doesn't merge in this repo - I'll check in tomorrow to see if someone else has merged this, and if not I'll try to escalate for you. Thanks!

**Comment by lym953** [comment]
Confirmed with @purple4reina and @TophrC-dd that this PR is good. Merging.

**Comment by lym953** [comment]
Merged and released.

**Comment by samchungy** [comment]
Appreciate it, thanks!

**Pull request merged**